### PR TITLE
build(app-shell): include standalone Python 3.10 with app

### PR DESCRIPTION
**Pull request by mcous**
## Overview

This PR builds a standalone Python 3.10 executable, with the `opentrons` package installed, into the Opentrons App.

### More work

Areas for future (or present, if you can figure out how to get it in this PR) improvement:

- Long build time on macOS
- Larger bundle size
- No messaging to user if Python environment can't be found or used 

## Changelog

- Download and include standalone Python in app-shell build
- Install local `opentrons` and `opentrons-shared-data` packages into the standalone Python environment
- Update CI configs to enable the build changes

## Review requests

Smoke test production assets built by CI. Ensure:

- App logs are showing the correct Python environment getting selected
- Local protocol analysis works
- No security warnings get triggered

We need to test this on all three OS's:

- [ ] Windows
- [ ] macOS
- [ ] Linux

## Risk assessment

High. We need to make sure code-signing goes well on these builds, or else macOS's Gatekeeper or Windows anti-virus programs might get pretty mad about this.

### CODE REVIEW

File: app-shell/package.json [hunk 1]
```diff
@@ -22,11 +22,6 @@
     "url": "https://github.com/Opentrons/opentrons/issues"
   },
   "homepage": "https://github.com/Opentrons/opentrons",
-  "workspaces": {
-    "nohoist": [
-      "**"
-    ]
-  },
```
Comment (mcous) [inline app-shell/package.json]: Unintentional, should no be removed

File: app-shell/src/python/index.ts [hunk 2]
```diff
@@ -36,37 +38,41 @@ function findPython(): string | undefined {
       fs.accessSync(path, fs.constants.X_OK)
       const stats = fs.statSync(path)
       if (stats.isFile()) {
-        log.debug('Python candidate selected', { path })
-        return path
+        log.info('Python environment selected', { path })
+        pythonPath = path
```
Comment (mcous) [inline app-shell/src/python/index.ts]: This will break the loop's early return, which we need

File: .github/workflows/app-test-build-deploy.yaml [hunk 3]
```diff
@@ -97,6 +97,9 @@ jobs:
       - uses: 'actions/setup-node@v1'
         with:
           node-version: '14'
+      - uses: actions/setup-python@v3
```
Comment (shlokamin) [inline .github/workflows/app-test-build-deploy.yaml:100]: 😮

File: app-shell/scripts/before-build.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+  return download(standalonePython.url)
+    .then(data => {
+      const hasher = crypto.createHash('sha256')
+      hasher.update(data)
+      const downloadHash = hasher.digest('hex')
+
+      if (downloadHash !== sha256) {
+        throw new Error(
+          `SHA265 mismatch; expected ${sha256}, got ${downloadHash}`
```
Comment (b-cooper) [inline app-shell/scripts/before-build.js]:
```suggestion
          `SHA256 mismatch; expected ${sha256}, got ${downloadHash}`
```

**Review by b-cooper — APPROVED** [review]
These changes all look good to me.  One quick comment typo. 
🐍 🍎

**Review by shlokamin — APPROVED** [review]
LGTM!

File: package.json [hunk 5]
```diff
@@ -80,6 +80,8 @@
     "cypress": "^6.6.0",
     "cypress-file-upload": "3.5.3",
     "cz-conventional-changelog": "2.1.0",
+    "decompress": "^4.2.1",
+    "download": "^8.0.0",
```
Comment (shlokamin) [inline package.json]: should we pin these dependencies to minimize variability?

Comment (b-cooper) [inline package.json]:
```suggestion
    "decompress": "4.2.1",
    "download": "8.0.0",
```

**Pull request merged**